google/cloud/spanner_dbapi/cursor.py

```diff
@@ -178,6 +178,8 @@ def execute(self, sql, args=None):
                     ddl = ddl.strip()
                     if ddl:
                         self.connection._ddl_statements.append(ddl)
+                if self.connection.autocommit:
```

Comment:
I think we should execute the statements immediately regardless of whether `autocommit` is set or not. Because Spanner handles DDL separately from transactions, there's no way to roll back DDL statements (which we will need to clearly document) and so I think having different behavior based on `autocommit` is unnecessary.

@AVaksman WDYT? Note that we now support executing multiple DDL statements at once which should alleviate the batching concern.

Reply:
@larkee, no problem, I'll delete the line.

I'm writing a system test to check the case.

Also I found out that in !autocommit DDLs are also not executed, even if one will call `commit()`. Will fix it in this PR as well.

![Безымянный](https://user-images.githubusercontent.com/26030071/112282589-4d04a600-8c98-11eb-9a25-0065616656ce.png)

Reply:
@larkee, you know, about regarding autocommit mode... I think users don't usually go deep into frameworks code. So, we support DDLs separated with `;`, still users may use frameworks, which are generating and executing DDLs one by one. In this case it's probably better to look after `autocommit` mode, to make DDLs pumping up in !autocommit mode - to execute them all at once little bit later.

Reply:
Hmm, that's a good point. In that case, we'll need to add `run_prior_DDL_statements()` to `commit()` as currently the statements aren't being executed even when `commit()` is called.